### PR TITLE
CVE-2010-4574 and CVE-2015-6757

### PR DESCRIPTION
**Pull request by SnowBirdMV**

### CODE REVIEW

File: cves/CVE-2015-6757.yml [hunk 1]
```diff
@@ -11,8 +11,10 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
 announced_instructions: |
+  A note for grading, I looked over this vulnerability and everything was
```
Comment (andymeneely) [inline cves/CVE-2015-6757.yml]: Makes sense - but make it better!! Don't just leave it alone.

File: cves/CVE-2015-6757.yml [hunk 2]
```diff
@@ -105,7 +108,7 @@ discovered:
   date: 2015-09-09
   automated: true
   google: false
-  contest: 
+  contest: blank
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-6757.yml]: Set to "false" if not found via contest

File: cves/CVE-2010-4574.yml [hunk 3]
```diff
@@ -32,15 +33,19 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  The pickle funciton in google chrome and chrome OS perormed pointer arithmetic
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-4574.yml]: Fix spelling

File: cves/CVE-2010-4574.yml [hunk 4]
```diff
@@ -54,14 +59,14 @@ fixes_vcc_instructions: |
...
-- :commit: 2ac837a92d049c5bf89b77888ca2c80be0f260c2
-  :note: ''
+  :note: |
+    Http Cache: Add some debug code to track a bug. 
+    Basically just an early check for the headers.
 vccs:
-- :commit:
-  :note: 
+- :commit: 2ac837a92d049c5bf89b77888ca2c80be0f260c2
+  :note: note
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-4574.yml]: You can leave this blank ('') if the commit msg was empty

File: cves/CVE-2010-4574.yml [hunk 5]
```diff
@@ -82,9 +87,10 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    There was no unit tests for this vulnerability.
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-4574.yml]: Contradictory response here: fields are both marked true, but answer says no tests were involved?

File: cves/CVE-2010-4574.yml [hunk 6]
```diff
@@ -100,20 +106,25 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    This vulnerabiloity was discovered througha bug report filed here
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-4574.yml]: Spelling

File: cves/CVE-2010-4574.yml [hunk 7]
```diff
@@ -171,8 +182,10 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      Input in some instances on 64 bit systems was trusted and led to incorrect
+      pointer arithmitic.  This was solved through imporved sanatization.
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-4574.yml]: Spelling

File: cves/CVE-2015-6757.yml [hunk 8]
```diff
@@ -83,7 +85,8 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
+  answer: |
+    There were unit tests for this vulnerability
```
Comment (CatherineOsadciw) [inline cves/CVE-2015-6757.yml]: What kinds of tests?

File: cves/CVE-2015-6757.yml [hunk 9]
```diff
@@ -11,8 +11,10 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: true
```
Comment (dxc7790) [inline cves/CVE-2015-6757.yml]: In the description, it talks about freeing a service worker before it was done being used. In what context was this happening? Why was the service worker freed before it was done? 

+1 Upvote

File: cves/CVE-2010-4574.yml [hunk 10]
```diff
@@ -203,4 +216,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    Coding mistakes.  The issue that caused this is easily overlooked in the
```
Comment (olucomedy) [inline cves/CVE-2010-4574.yml]: Could you give an example or two of the coding mistakes that were made.

File: cves/CVE-2015-6757.yml (hunk 2 above)
Comment (SnowBirdMV) [reply cves/CVE-2015-6757.yml]: fixed in fe7997d1efd34e7fa979fdd2760789e719ee96a3

File: cves/CVE-2010-4574.yml (hunk 4 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in fe7997d1efd34e7fa979fdd2760789e719ee96a3

File: cves/CVE-2010-4574.yml (hunk 6 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in fe7997d1efd34e7fa979fdd2760789e719ee96a3

File: cves/CVE-2010-4574.yml (hunk 5 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in fe7997d1efd34e7fa979fdd2760789e719ee96a3

File: cves/CVE-2010-4574.yml (hunk 7 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in fe7997d1efd34e7fa979fdd2760789e719ee96a3

File: cves/CVE-2010-4574.yml (hunk 3 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in fe7997d1efd34e7fa979fdd2760789e719ee96a3

File: cves/CVE-2010-4574.yml (hunk 10 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in cd6228e51c1be5a5aceb5ebed9c887c15e66ac41

File: cves/CVE-2015-6757.yml (hunk 9 above)
Comment (SnowBirdMV) [reply cves/CVE-2015-6757.yml]: fixed in cd6228e51c1be5a5aceb5ebed9c887c15e66ac41

File: cves/CVE-2015-6757.yml (hunk 8 above)
Comment (SnowBirdMV) [reply cves/CVE-2015-6757.yml]: fixed in cd6228e51c1be5a5aceb5ebed9c887c15e66ac41

File: cves/CVE-2010-4574.yml [hunk 11]
```diff
@@ -100,20 +108,25 @@ discovered:
...
 subsystem:
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
+  answer: |
+    from the blog post
```
Comment (MaeveMonster) [inline cves/CVE-2010-4574.yml]: I'd suggest making this a complete sentence for better readability.

File: cves/CVE-2010-4574.yml [hunk 12]
```diff
@@ -69,7 +74,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 1
```
Comment (MaeveMonster) [inline cves/CVE-2010-4574.yml]: add 3 upvotes from me

Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: added in bc7f9648f9213c4625e4cd4da36dda131e0dd98b

File: cves/CVE-2010-4574.yml (hunk 11 above)
Comment (SnowBirdMV) [reply cves/CVE-2010-4574.yml]: fixed in bc7f9648f9213c4625e4cd4da36dda131e0dd98b

**Pull request merged**